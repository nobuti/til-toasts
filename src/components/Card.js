/** @jsx jsx */
import { useContext } from 'react'
import { jsx, css } from '@emotion/core'
import styled from '@emotion/styled'

import ToastContext from './toastContext'
import Button from './form/Button'
import { factory } from './Toast'

import { ReactComponent as Atom } from '../assets/atom.svg'
import { ReactComponent as Ghost } from '../assets/ghost.svg'
import { ReactComponent as Virus } from '../assets/virus.svg'
import { ReactComponent as Lego } from '../assets/lego.svg'
import { ReactComponent as Heart } from '../assets/heart.svg'

const Action = styled(Button)`
  width: 64px;
  height: 64px;
  border-radius: 50%;
  padding: 0;

  svg {
    width: 48px;
    height: 48px;
    stroke-width: 1.25;
  }
`

const Card = () => {
  const { addToast } = useContext(ToastContext)

  return (
    <ul
      css={css`
        display: flex;

        li + li {
          margin-left: 24px;
        }
      `}
    >
      <li>
        <Action
          css={css`
            color: #2548f5;
          `}
          onClick={() => addToast(factory('atom'))}
        >
          <Atom />
        </Action>
      </li>

      <li>
        <Action
          css={css`
            color: #f58325;
          `}
          onClick={() => addToast(factory('ghost'))}
        >
          <Ghost />
        </Action>
      </li>

      <li>
        <Action
          css={css`
            color: #f5eb25;
          `}
          onClick={() => addToast(factory('virus'))}
        >
          <Virus />
        </Action>
      </li>

      <li>
        <Action
          css={css`
            color: #e425f5;
          `}
          onClick={() => addToast(factory('lego'))}
        >
          <Lego />
        </Action>
      </li>

      <li>
        <Action
          css={css`
            color: #f52537;
          `}
          onClick={() => addToast(factory('heart'))}
        >
          <Heart />
        </Action>
      </li>
    </ul>
  )
}

export default Card
