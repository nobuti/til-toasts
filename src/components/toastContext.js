import React, { createContext } from 'react'

import useToast from '../hooks/useToast'
const ToastContext = createContext(null)

export const ToastProvider = ({ children }) => {
  const toast = useToast()
  return <ToastContext.Provider value={toast}>{children}</ToastContext.Provider>
}

export default ToastContext
