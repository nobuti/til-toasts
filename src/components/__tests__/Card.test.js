import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import Card from '../Card'
import * as ToastContext from '../toastContext'

describe('Card', () => {
  const addToast = jest.fn()
  jest
    .spyOn(ToastContext, 'useToastContext')
    .mockImplementation(() => ({ addToast }))

  it('should render properly', () => {
    const { container } = render(<Card />)
    expect(container).toMatchSnapshot()
  })

  it('should create toasts on click', async () => {
    const { container } = render(<Card />)
    const button = container.querySelector('button')

    fireEvent.click(button, {
      target: button,
    })

    expect(addToast).toHaveBeenCalled()
  })
})
