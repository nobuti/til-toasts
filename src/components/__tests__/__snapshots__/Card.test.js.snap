// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render properly 1`] = `
<div>
  <ul
    class="css-11tbn42-Card"
  >
    <li>
      <button
        class="css-19tdhsw-Card"
      >
        <svg>
          atom.svg
        </svg>
      </button>
    </li>
    <li>
      <button
        class="css-nb07tz-Card"
      >
        <svg>
          ghost.svg
        </svg>
      </button>
    </li>
    <li>
      <button
        class="css-15nwj7g-Card"
      >
        <svg>
          virus.svg
        </svg>
      </button>
    </li>
    <li>
      <button
        class="css-yfhoj6-Card"
      >
        <svg>
          lego.svg
        </svg>
      </button>
    </li>
    <li>
      <button
        class="css-1fus6uf-Card"
      >
        <svg>
          heart.svg
        </svg>
      </button>
    </li>
  </ul>
</div>
`;
